### PR TITLE
[CARBONDATA-3071] Add CarbonSession Java Example

### PR DESCRIPTION
**Pull request by lijiang2**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed? NA
 
 - [ ] Any backward compatibility impacted? NA
 
 - [ ] Document update required? NA

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by lijiang2** [comment]
@jackylk PTAL.

**Comment by chenliang613** [comment]
add to whitelist

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/780/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/977/

File: examples/spark2/src/main/java/org/apache/carbondata/examples/sql/CarbonSessionExample.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+public class CarbonSessionExample {
+
+  public static void main(String[] args) {
+    File file = new File(CarbonSessionExample.class.getResource("/").getPath() + "../..");
+    try {
+      String rootPath = file.getCanonicalPath();
+      System.setProperty("path.target", rootPath + "/examples/spark2/target");
+      PropertyConfigurator.configure(rootPath + "/examples/spark2/src/main/resources/log4j.properties");
+      CarbonProperties.getInstance().addProperty(CarbonCommonConstants.ENABLE_QUERY_STATISTICS, "true");
+      SparkSession spark = ExampleUtils.createCarbonSession("JavaCarbonSessionExample", 1);
```
Comment (xuchuanyin) [inline examples/spark2/src/main/java/org/apache/carbondata/examples/sql/CarbonSessionExample.java]: I think you can change this variable name to 'carbon' to tell the user that it is not a raw SparkSession.

Besides, I think it's better to add an example for DataFrame  (write dataframe to carbon table).

File: examples/spark2/src/main/java/org/apache/carbondata/examples/sql/CarbonSessionExample.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                      "bigintField LONG, " +
+                      "doubleField DOUBLE, " +
+                      "stringField STRING, " +
+                      "timestampField TIMESTAMP, " +
+                      "decimalField DECIMAL(18,2), " +
+                      "dateField DATE, " +
+                      "charField CHAR(5), " +
+                      "floatField FLOAT " +
+                      ") " +
+                      "STORED BY 'carbondata' " +
```
Comment (jackylk) [inline examples/spark2/src/main/java/org/apache/carbondata/examples/sql/CarbonSessionExample.java]: use `STORED AS carbondata` instead

**Comment by jackylk** [comment]
Hi, are you still working on this PR?

**Comment by lijiang2** [comment]
Thank you for the suggestion. Yes, I will work on it soon.

**Comment by jackylk** [comment]
Thanks, and you can create a JIRA ticket by https://issues.apache.org/jira/projects/CARBONDATA/issues, then you can modify the PR title to include the JIRA number

File: examples/spark2/src/main/java/org/apache/carbondata/examples/sql/CarbonSessionExample.java (hunk 1 above)
Comment (lijiang2) [reply examples/spark2/src/main/java/org/apache/carbondata/examples/sql/CarbonSessionExample.java]: Thank you for the suggestion. I updated the variable name. And I think I will start the DataFrame example in another pr.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1252/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1467/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9516/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**